.github/workflows/ci.yml

```diff
@@ -0,0 +1,119 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches: main
+  merge_group:
+
+permissions: {}
+
+jobs:
+  test-msrv-required:
+    name: Test ${{ matrix.state }} on ${{ matrix.target }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        target:
+          - Linux
+        state:
+          - default
+          - NU7
+
+        include:
+          - target: Linux
+            os: ubuntu-latest
+
+          - state: NU7
+            rustflags: '--cfg zcash_unstable="nu7"'
+
```

Comment:
Nit: I find it difficult to see where one job begins and ends when there are whitespace lines internally; indentation is sufficient to separate these sections for me. If we want to have whitespace lines internal to a job, could we have a comment line with a separator or something like that that makes each independent job more visually distinguishable?